ui/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -185,9 +186,11 @@ const mapStateToProps = (state, ownProps) => {
   );
 
   const isNftTransfer = Boolean(
-    allNftContracts?.[selectedAddress]?.[chainId]?.find((contract) => {
-      return isEqualCaseInsensitive(contract.address, fullTxData.txParams.to);
-    }),
+    allNftContracts?.[selectedAddress]?.[hexToDecimal(chainId)]?.find(
```

Comment:
Why is this change necessary?

Reply:
has `chainId` changed to a hex value? or was this buggy before?

Reply:
This was buggy before... its an unfortunate effect of the situation where the two network controllers (extension and core - which will be merged soon!) represent the chainId differently (extension as a hex, and core as a decimal). This results in a bunch of other controllers from core expecting chainIds as decimals. This includes `NftController`, so any time we read state from this controller we have to convert the chainId to decimal. related: https://github.com/MetaMask/metamask-extension/pull/17384